code/modules/reagents/chemistry/holder.dm

```diff
@@ -66,6 +65,12 @@
 	var/list/datum/reagent/addiction_list
 	/// various flags, see code\__DEFINES\reagents.dm
 	var/flags
+	///list of reactions currently on going
+	var/list/datum/equilibrium/reaction_list = new/list()
+	///cached list of reagents
+	var/list/datum/reagent/previous_reagent_list = new/list()
```

Comment:
```suggestion
	var/list/datum/equilibrium/reaction_list = list()
	///cached list of reagents
	var/list/datum/reagent/previous_reagent_list = list()
```
Just use the `list()` initializer for lists, as we do in most of the (not ancient) code.
Consider making them lazy lists if there will be many empty ones.

Reply:
To make them lazylists, I leave it like this, right? 
```
var/list/datum/equilibrium/reaction_list
var/list/datum/reagent/previous_reagent_list
```

Reply:
Correct, and then you use the lazy list macros to manipulate them (`LAZYADD()`, `LAZYREMOVE()` and so on).

Reply:
This should be done now